fixes #8892, #9340 - Validates content label and override value, BZ1173723, BZ1187750

cfouant opened the pull request.

jlsherrill commented on app/controllers/katello/api/v2/activation_keys_controller.rb:
```diff
@@ -206,23 +206,28 @@ def remove_subscriptions
+    param :content_override, Hash, :desc => N_("Content override parameters") do
+      param :content_label, String, :desc => N_("Label of the content"), :required => true
+      param :value, :bool, :desc => N_("Override to 0/1, or 'default'"), :required => true
+    end
```
would this api change affect hammer at all?  what was the reasoning behind making the api change (versus keeping the parameters the same)?

cfouant replied:
I actually didn't realize hammer could handle nested params. I've tested it in hammer and it seems to work, and removing the different handling of the two ways to handle cleans the code up a bit.

cfouant commented:
@jlsherrill - Hammer is able to handle the nested params, which I was unaware of previously. So to make everything more uniform, I removed the different handling of the calls between CLI, API and UI. I also updated the UI to keep everything handled more uniformly as well (call in 'default' rather than null).

jlsherrill commented:
ACK
